python_modules/dagster/dagster/core/execution/resources_init.py

```diff
@@ -110,7 +110,7 @@ def _core_resource_initialization_event_generator(
     pipeline_def_for_backwards_compat: Optional[PipelineDefinition],
 ):
 
-    pipeline_name = None
+    pipeline_name = ""
```

Comment:
maybe worth a more obnoxious placeholder here since use and setting real value are gated by `emit_persistent_events`, so not expected to be used 

alternatively consider leaving a comment here highlighting awkward setup